### PR TITLE
Add more logging info to storage and reader/writer open operations

### PR DESCRIPTION
**Pull request by MichaelOrlov**
This PR adds more logging information in cases when the reader or writer fails to open bag files.
- Added check if `storage.uri` is an empty string
- Print out a list of available storage plugins in case of failure to open a bag file
- Print out the value of the provided `storage.uri` if fail to open a bag file.

### CODE REVIEW

File: rosbag2_storage/src/rosbag2_storage/impl/storage_factory_impl.hpp [hunk 1]
```diff
@@ -17,8 +17,10 @@
 #define ROSBAG2_STORAGE__IMPL__STORAGE_FACTORY_IMPL_HPP_
 
 #include <algorithm>
+#include <vector>
 #include <memory>
 #include <string>
+#include <unordered_set>
```
Comment (christophebedard) [inline rosbag2_storage/src/rosbag2_storage/impl/storage_factory_impl.hpp]: move `<vector>` to the end:

```suggestion
#include <algorithm>
#include <memory>
#include <string>
#include <unordered_set>
#include <vector>
```

File: rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp [hunk 2]
```diff
@@ -108,7 +112,8 @@ void SequentialReader::open(
   } else {
     storage_ = storage_factory_->open_read_only(storage_options_);
     if (!storage_) {
-      throw std::runtime_error{"No storage could be initialized from the inputs."};
+      throw std::runtime_error("No storage could be initialized for the input uri: " +
```
Comment (christophebedard) [inline rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp]: for consistency:

```suggestion
      throw std::runtime_error("No storage could be initialized for the input URI: " +
```

**Comment by christophebedard** [comment]
Looks good to me other than some minor comments.

**Review by fujitatomoya — APPROVED** [review]
i do not have any objections for this. lgtm with @christophebedard 's comments.

File: rosbag2_storage/src/rosbag2_storage/impl/storage_factory_impl.hpp (hunk 1 above)
Comment (MichaelOrlov) [reply rosbag2_storage/src/rosbag2_storage/impl/storage_factory_impl.hpp]: Fixed in the https://github.com/ros2/rosbag2/pull/1881/commits/505035c289f76d66d030946ab6146427c6827d1a

File: rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp (hunk 2 above)
Comment (MichaelOrlov) [reply rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp]: Fixed in the https://github.com/ros2/rosbag2/pull/1881/commits/505035c289f76d66d030946ab6146427c6827d1a

**Comment by MichaelOrlov** [comment]
Pulls: ros2/rosbag2#1881
Gist: https://gist.githubusercontent.com/MichaelOrlov/d7c7dbb089647582fbca4052a7712f23/raw/5a2496f5d65a874725cdd150a6a6994e0af561b2/ros2.repos
BUILD args:  --packages-above-and-dependencies rosbag2_cpp rosbag2_storage rosbag2_tests
TEST args:  --packages-above rosbag2_cpp rosbag2_storage rosbag2_tests
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/14939
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/22251/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/16511/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1773/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/23029/)

**Review by christophebedard — APPROVED** [review]

**Comment by MichaelOrlov** [comment]
https://github.com/Mergifyio backport jazzy

**Pull request merged**